Source/WebKit/Configurations/NetworkService.xcconfig

```diff
@@ -28,10 +28,11 @@ PRODUCT_BUNDLE_IDENTIFIER = $(PRODUCT_NAME);
 INFOPLIST_FILE[sdk=embedded*] = NetworkProcess/EntryPoint/Cocoa/XPCService/NetworkService/Info-iOS.plist;
 INFOPLIST_FILE[sdk=macosx*] = NetworkProcess/EntryPoint/Cocoa/XPCService/NetworkService/Info-OSX.plist;
 
-OTHER_LDFLAGS = $(inherited) $(OTHER_LDFLAGS_STAGED_FRAMEWORK_PATH) $(WK_RELOCATABLE_FRAMEWORKS_LDFLAGS) $(SOURCE_VERSION_LDFLAGS) $(PROFILE_GENERATE_OR_USE_LDFLAGS);
-OTHER_LDFLAGS[sdk=macosx*] = $(inherited) $(SOURCE_VERSION_LDFLAGS);
+// Override values in BaseTarget.xcconfig to disable PGO profile generation.
+PROFILE_GENERATE_OR_USE_CFLAGS_ENABLE_LLVM_PROFILE_GENERATION[sdk=macosx*] = ;
+PROFILE_GENERATE_OR_USE_LDFLAGS[sdk=macosx*] = ;
```

Comment:
It feels wrong that we are disabling PGO _generation_ but not _usage_ (i.e. we aren't stubbing out all of PROFILE_GENERATE_OR_USE_CFLAGS, only the enable-generation branch). This means we are building the XPC services with PGO profiles that are based on instrumented framework code. It's probably not causing a performance loss, but it's unnecessary.

Similarly it feels wrong that PGO is completely enabled in GPUService.xcconfig.

I know this is not your bug, but since you're refactoring this, could you fix these issues and move all the PGO-disabling logic to BaseXPCService.xcconfig?

Reply:
Yep, I can do this, but I have other tasks that are more important, so I don't have time to spend on this now.  (This was something I started to fix as part of <https://github.com/WebKit/WebKit/pull/3811>, but decided to split it out into its own PR.)

Will come back to it later.